tests/core/test_otel_logger.py

```diff
@@ -356,3 +356,110 @@ def test_timer_start_and_stop_manually_send_true(self, mock_time, metrics_consis
         self.meter.get_meter().create_observable_gauge.assert_called_once_with(
             name=full_name(name), callbacks=ANY
         )
+
+    def test_incr_counter(self):
+        # Arrange
+        logger = self.stats
```

Comment:
Nitpick: the term "logger" is a bit overloaded in Airflow and this could be misread as logger meaning the console like log.error().  Consider renaming this (here and below) to `otel` or `otel_logger` just to help clarify, or just using `self.stats` directly.